src/utils.py

```diff
@@ -0,0 +1,215 @@
+    metadata_file = f"views/aggregate-{variety}-metadata.json"
+
+    # read the existing flow metadata from existing JSON file
+    github_token = Secret.load("github-token")
+    gh = github3.login(token=github_token.get())
+    repo = gh.repository("PrefectHQ", repo_name)
+    collection_repo = gh.repository("PrefectHQ", collection_name)
+    latest_release = collection_repo.latest_release().tag_name
+
+    existing_metadata_content = repo.file_contents(
+        metadata_file, ref=branch_name
+    ).decoded.decode()
+
+    existing_metadata_dict = json.loads(existing_metadata_content)
+
+    existing_metadata_dict.update(collection_metadata)
+
+    updated_metadata_dict = dict(sorted(existing_metadata_dict.items()))
+
+    # create a new commit adding the collection version metadata
+    try:
+        repo.create_file(
+            path=f"{variety}s/{collection_name}/{latest_release}.json",
+            message=f"Add {collection_name} {latest_release} to {variety} records",
+            content=json.dumps(collection_metadata, indent=4).encode("utf-8"),
+            branch=branch_name,
+        )
+    except github3.exceptions.UnprocessableEntity as e:
+        if '"sha" wasn\'t supplied' in str(e):
+            # file already exists so nothing to update
+            print(
+                f"{variety} metadata for {collection_name} {latest_release} already exists!"
+            )
+            return
+        else:
+            raise
+
+    # create a new commit updating the aggregate flow metadata file
+    updated_metadata_content = json.dumps(updated_metadata_dict, indent=4)
+    if existing_metadata_content == updated_metadata_content:
+        print(
+            f"Aggregate {variety} metadata for {collection_name} {latest_release} already up to date!"
+        )
+        return
+
+    repo.file_contents(metadata_file, ref=branch_name).update(
+        message=f"Update aggregate {variety} metadata with {collection_name} {latest_release}",
+        content=updated_metadata_content.encode("utf-8"),
+        branch=branch_name,
+    )
+
+    print(
+        f"Updated aggregate {variety} metadata for {collection_name} {latest_release}!"
+    )
+
+
+def get_collection_names():
+    catalog_resp = httpx.get("https://docs.prefect.io/collections/catalog/")
+    catalog_soup = BeautifulSoup(catalog_resp.text, "html.parser")
+    return sorted(
+        {
+            url.split("/")[-2]
+            for div in catalog_soup.find_all("div", class_="collection-item")
+            if "prefecthq.github.io" in (url := div.find("a")["href"])
+            and url not in exclude_collections
+        }
+    )
+
+
+def get_logo_url_for_collection(collection_name: str) -> str:
+    """Returns the URL of the logo for a collection."""
+
+    blocks_metadata = read_view_content("block")
+
+    block_types_from_collection = blocks_metadata[collection_name]["block_types"]
+
+    return block_types_from_collection.popitem()[1]["logo_url"]
```

Comment:
is there another place we keep logo urls besides Blocks? it seems ideal (in this case) to have a logo url be associated with the collection instead of the blocks inside, but not sure if that association exists anywhere